src/smc-project/gofmt.ts

```diff
@@ -0,0 +1,78 @@
+const { writeFile, readFile } = require("fs");
+const tmp = require("tmp");
+const { callback } = require("awaiting");
+const { spawn } = require("child_process");
+// const { replace_all } = require("smc-util/misc");
+
+interface ParserOptions {
+  parser: string;
+  tabWidth: number;
+}
+
+function close(proc, cb): void {
+  proc.on("close", code => cb(undefined, code));
+}
+
+// ref: https://golang.org/cmd/gofmt/ ... but there isn't anything to configure
+
+function run_gofmt(input_path: string) {
+  const args = ["-w", input_path];
+  return spawn("gofmt", args);
+}
+
+function cleanup_error(err: string, tmpfn: string): string {
+  let ret: string[] = [];
+  for (let line of err.split("\n")) {
+    if (line.startsWith(tmpfn)) {
+      line = line.slice(tmpfn.length + 1);
+    }
+    ret.push(line);
+  }
+  return ret.join("\n");
+}
+
+export async function gofmt(
+  input: string,
+  options: ParserOptions,
+  logger: any
+): Promise<string> {
+  // create input temp file
+  const input_path: string = await callback(tmp.file);
```

Comment:
Put the body of this function in a try/finally block, and delete input_path in the finally part.  As far as I can tell reading this code, input_path is never cleaned up at all.      If you don't want to indent everything in a try/finally, move the body of code to a helper function.

Reply:
Alternatively, since gofmt takes input from stdin by default, why not just write to stdin?  If it works, shouldn't it be more efficient than creating a temporary file?    E.g., there's an example here https://stackoverflow.com/questions/13230370/nodejs-child-process-write-to-stdin-from-an-already-initialised-process

Reply:
(If you do use stdin, be sure to test on a file that's big-ish e.g., at least 40K or so, to make sure there is no weird locking/buffering messing things up.)